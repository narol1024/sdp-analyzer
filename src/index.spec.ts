/**
 * This is a sample test suite.
 * Replace this with your implementation.
 */
import { analyze } from './index';

describe('Main Tests', () => {
  it('can analyze a valid npm package', async () => {
    await expect(analyze('react')).resolves.toMatchSnapshot();
  });
  it('can analyze for local packages', async () => {
    await expect(analyze('./src/platforms/workspaces/yarn/fixture')).resolves.toMatchSnapshot();
  });
});
