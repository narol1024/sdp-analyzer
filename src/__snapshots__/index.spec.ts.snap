// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Tests can analyze a valid npm package 1`] = `
Array [
  Object {
    "fanIn": 10000,
    "fanOut": 1,
    "label": "Stable",
    "name": "react",
    "stability": 0.00009999000099990002,
  },
]
`;

exports[`Main Tests can analyze for local packages 1`] = `
Array [
  Object {
    "fanIn": 1,
    "fanOut": 0,
    "instable": 0,
    "label": "Stable",
    "name": "a",
  },
  Object {
    "fanIn": 1,
    "fanOut": 1,
    "instable": 0.5,
    "label": "Normal",
    "name": "b",
  },
  Object {
    "fanIn": 1,
    "fanOut": 2,
    "instable": 0.6666666666666666,
    "label": "Flexible",
    "name": "c",
  },
  Object {
    "fanIn": 1,
    "fanOut": 3,
    "instable": 0.75,
    "label": "Flexible",
    "name": "d",
  },
  Object {
    "fanIn": 1,
    "fanOut": 4,
    "instable": 0.8,
    "label": "Instable",
    "name": "e",
  },
]
`;
